docs/sql-data-sources-jdbc.md

```diff
@@ -29,7 +29,9 @@ as a DataFrame and they can easily be processed in Spark SQL or joined with othe
 The JDBC data source is also easier to use from Java or Python as it does not require the user to
 provide a ClassTag.
 (Note that this is different than the Spark SQL JDBC server, which allows other applications to
-run queries using Spark SQL).
+run queries using Spark SQL). 
+
+All columns are automatically converted to be nullable for compatibility reasons.
```

Comment:
I actually think this is something we should fix but we couldn't because it is a too much breaking change. This is not only for JDBC but for other filed based sources.

We should better have an option or configuration to set the nullability correctly, and make it disabled by default.

Reply:
@HyukjinKwon Yes, agree, we should find better solution for this  'nullable' issue. But until we fix, adding a note in the document will be helpful to the users.

Reply:
+1 on what @HyukjinKwon suggested.